implementations/elixir/ockam/ockam/lib/ockam/transport/udp/listener.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Ockam.Transport.UDP.Listerner do
+  @moduledoc false
+
+  use Ockam.Worker
+
+  alias Ockam.Message
+  alias Ockam.Transport.UDPAddress
+  alias Ockam.Wire
+
+  @wire_encoder_decoder Ockam.Wire.Binary.V1
+
+  @doc false
+  @impl true
+  def setup(options, state) do
+    ip = Keyword.get_lazy(options, :ip, &default_ip/0)
+    port = Keyword.get_lazy(options, :port, &default_port/0)
+
+    udp_open_options = [:binary, :inet, {:ip, ip}, {:active, true}]
+    udp_address = %UDPAddress{ip: ip, port: port}
+
+    route_outgoing = Keyword.get(options, :route_outgoing, false)
+
+    with {:ok, socket} <- :gen_udp.open(port, udp_open_options),
+         :ok <- setup_routed_message_handler(route_outgoing, state.address) do
+      state = Map.put(state, :socket, socket)
+      state = Map.put(state, :udp_address, udp_address)
+      {:ok, state}
+    end
+  end
+
+  defp setup_routed_message_handler(true, listerner) do
+    handler = fn message -> handle_routed_message(listerner, message) end
+
+    with :ok <- Router.set_message_handler(2, handler),
```

Comment:
what's `2`?

Reply:
thanks, its the udp address type in our binary protocol
I'll add module constant 👍

Reply:
fixed